### PR TITLE
Added Date-Only Feature

### PR DESCRIPTION
**Pull request by MarketingPip**
## Summary

Added option for date-only. As outlined in your instructions as mentioned in issue #58
 
Again, you will need to do unit test's as I am not familiar with the process. But I have done manual tests for this on my end. 

This time I have added the details to the README files. 

<br>

> Note: I added a step to make Python file executable in Dockerfile. As when making changes to script on my end - somewhere file permissions got changed & Dockerfile would NOT run properly. This will prevent any PR's that do NOT have proper file permission's wrote from not executing properly inside the container. 

<br>

ps; thank you for helping me improve my future PR's :heart: 
(still learning my way around GitHub)

## Closing Issues

This close's the issue / feature request as mentioned in issue #58 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Improvements to existing code, such as refactoring or optimizations (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] I have read the [**CONTRIBUTING**](https://github.com/cicirello/.github/blob/main/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: generatesitemap.py [hunk 1]
```diff
@@ -298,7 +301,8 @@ def writeXmlSitemap(files, baseUrl, dropExtension=False) :
     sitemapFormat = sys.argv[5]
     additionalExt = set(sys.argv[6].lower().replace(",", " ").replace(".", " ").split())
     dropExtension = sys.argv[7]=="true"
-
+    date_only = sys.argv[8]
```
Comment (cicirello) [inline generatesitemap.py:304]: There's currently a merge conflict due to some refactoring I was in the middle of, and just merged. However, this comment here is still applicable. Specifically, take a look at how the other boolean inputs are converted here from the strings that the Actions framework passes via command line params to actual boolean values, with a string comparison here to the string "true". This enables cleaner logic elsewhere. See my comment somewhere above where this is used.

File: generatesitemap.py [hunk 2]
```diff
@@ -222,6 +222,9 @@ def lastmod(f) :
                     universal_newlines=True).stdout.strip()
     if len(mod) == 0 :
         mod = datetime.now().astimezone().replace(microsecond=0).isoformat()
+    if date_only != "false":
```
Comment (cicirello) [inline generatesitemap.py:225]: First see my other comment lower down in file where the argument is first processed. Once that is handled, date_only will be a boolean, so you can eliminate the `!= "false"`.

**Review by cicirello — CHANGES_REQUESTED** [review]

**Comment by MarketingPip** [comment]
@cicirello Hello!

Sorry for any delays I will have this all wrapped up by this weekend, 

hope your having a great weekend :+1: 

ps; I did find some other resources on site map formats (I will have to see if I can find them) so you can possibly expect another PR after I finish this one up. (if you think it's needed + if I can even find the resources again).

**Comment by MarketingPip** [comment]
@cicirello - hate to be a bothersome. 

But can I get some help regarding the boolen / input? I am running into issues for the past two hours over this & do not know if it's related to my coding or GitHub actions itself etc.. 

Been struggling with this longer than writing my previous PR lol.. 

After this is fixed / resolved. I will make a clean PR / close this + update with changes.


Let me know and I will tag you the branch I am working in. 

Hope all is well :+1:

> ps: sorry for the long over-waited PR.

**Comment by cicirello** [comment]
@MarketingPip take a look at where the existing inputs are initially processed (where the main function is called). e.g., you'll find:

```Python
if __name__ == "__main__" :
    main(
        websiteRoot = sys.argv[1],
        baseUrl = sys.argv[2],
        includeHTML = sys.argv[3].lower() == "true",
        includePDF = sys.argv[4].lower() == "true",
        sitemapFormat = sys.argv[5],
        additionalExt = set(sys.argv[6].lower().replace(",", " ").replace(".", " ").split()),
        dropExtension = sys.argv[7].lower() == "true"
    )
```

All of the arguments in `sys.argv` are strings, as are command line arguments in general. However, logically, a couple of them are intended as booleans. For example, `sys.argv[7]` controls whether or not the extension `.html` is kept in URLs. Rather than passing `sys.argv[7]` as a string to the main function for the parameter `dropExtension`, the above block is using a string comparison to the string "true" to get a boolean.

**Pull request closed without merging**